src/components/FolderCreationModal.jsx

```diff
@@ -42,29 +43,37 @@ const FolderCreationModal = ({
   const [sortedPagesData, setSortedPagesData] = useState(pagesData)
 
   const sortOptions = [
-    ... parentFolder 
-      ? [{
-        value: 'folder',
-        label: 'Original order',
-      }]
-      : [],
     {
       value: 'title',
       label: 'Name'
     }
   ]
 
-  const baseApiUrl = `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}${parentFolder ? `/collections/${parentFolder}` : ''}`
-
-  const saveHandler = async () => {
-    // TODO
-    console.log('saving')
-  }
+  useEffect(() => {
+    const sortedOrder = pagesData.concat().sort(sortFuncs['title'])
+    setSortedPagesData(sortedOrder)
+  }, [])
+
+  const { mutateAsync: saveHandler, isLoading } = useMutation(
+    () => moveFiles(siteName, [ ...selectedFiles ], slugifyCategory(title), parentFolder),
+    { onSuccess: () => {
+        const redirectUrl = `/sites/${siteName}/folder/${parentFolder ? `${parentFolder}/subfolder/${slugifyCategory(title)}` : slugifyCategory(title)}`
+        setRedirectToPage(redirectUrl)
+        setIsFolderCreationActive(false)
+      },
+      onError: (error) => {
+        if (error.response.status === 409) {
+          errorToast(`The name chosen is a protected folder name. Please choose a different name.`)
+        } else {
+          errorToast(`There was a problem trying to save your nav bar. ${DEFAULT_RETRY_MSG}`)
```

Comment:
The error toast message should be about folder creation and not nav bar

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/369/commits/319ab8186837bb72b6043d5247cfeb0cc6fb7942